File: learning_platform_api_gateway/internal/dto/lessonDto/createLesson.go
package lessonDto

import (
	"time"
)

type CreateLesson struct {
	BoardID    *int64      `json:"board_id"`
	MeetLink   *string     `json:"meet_link"`
	StartTime  time.Time   `json:"start_time"`
	Duration   int64       `json:"duration"`
	TutorID    int64       `json:"tutor_id"`
	MediaItems []MediaItem `json:"media_items"`
	UserIDs    []int64     `json:"user_ids"`
}

File: learning_platform_front/src/entities/lesson/types/createLesson.ts
export type CreateLessonMediaItemRequest = {
    s3_link: string
    s3_preview: string
    type: "IMAGE" | "VIDEO"
}

export type CreateLessonRequest = {
    board_id?: number | null
    meet_link?: string | null
    start_time: string
    duration: number
    tutor_id: number
    media_items?: CreateLessonMediaItemRequest[]
    user_ids: number[]
}
